packages/apps/reputation-oracle/server/src/modules/qualification/qualification.error.filter.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpStatus,
+  Logger,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+
+import {
+  QualificationError,
+  QualificationErrorMessage,
+} from './qualification.error';
+
+@Catch(QualificationError)
+export class QualificationErrorFilter implements ExceptionFilter {
+  private logger = new Logger(QualificationErrorFilter.name);
+  catch(exception: QualificationError, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const request = ctx.getRequest<Request>();
+    let status = HttpStatus.BAD_REQUEST;
+
+    if (
+      exception.message === QualificationErrorMessage.NOT_FOUND ||
+      exception.message === QualificationErrorMessage.NO_WORKERS_FOUND
```

Comment:
Question: will it break something if we change `QualificationErrorMessage.NO_WORKERS_FOUND` to return `422` instead of `404`? We had this logic before refactoring, but would be nice to change it if it's a quick win

Reply:
It shouldn't be a problem.
I'll change it.